app/main.py

```diff
@@ -115,16 +120,29 @@ def grab_secrets(self):
                   with open(output_path, 'w') as cert_file:
                       cert_file.write(self.cert_to_pem(cert.cer))
 
+    def dump_secret(self, secret):
+              value = base64.decodestring(secret.value)
+              if secret.tags is not None and 'file-encoding' in secret.tags:
+                     encoding = secret.tags['file-encoding']
+                     if encoding == 'base64':
+                                 value = base64.decodestring(value)
+
+              return value
+
     def dump_pfx(self, pfx, name):
-              import base64
               from OpenSSL import crypto
               p12 = crypto.load_pkcs12(base64.decodestring(pfx))
               pk = crypto.dump_privatekey(crypto.FILETYPE_PEM, p12.get_privatekey())
               cert = crypto.dump_certificate(crypto.FILETYPE_PEM, p12.get_certificate())
+              ca_certs = p12.get_ca_certificates()
               with open(os.path.join(self.keys_output_folder, name), 'w') as key_file:
                     key_file.write(pk)
               with open(os.path.join(self.certs_output_folder, name), 'w') as cert_file:
                     cert_file.write(cert)
+              if ca_certs is not None:
```

Comment:
can we maybe de-dup some code with merging the cetr + ca_certs to a list and then just iterating it?
e.g.:
```python
certs = [p12.get_certificate()] + (p12.get_ca_certificates() or [])
with open(os.path.join(self.certs_output_folder, name), 'w') as cert_file:
    for cert in certs:
        cert_file.write(crypto.dump_certificate(crypto.FILETYPE_PEM, cert))
```

Reply:
done